plugins/notifications-backend/src/service/db.ts

```diff
@@ -0,0 +1,85 @@
+import { Config } from '@backstage/config';
+
+import knex, { Knex } from 'knex';
+
+// initDB
+// creates DB client and tables
+export async function initDB(dbConfig: Config): Promise<Knex<any, any>> {
+  // create db client
+  const dbClient = knex({
+    client: dbConfig.getString('client'),
+    connection: {
+      host: dbConfig.getString('connection.host'),
+      port: dbConfig.getNumber('connection.port'),
+      user: dbConfig.getString('connection.user'),
+      password: dbConfig.getString('connection.password'),
+      database: 'backstage_plugin_notifications',
+    },
+  });
+
+  // create tables
+  if (!(await dbClient.schema.hasTable('messages'))) {
+    await dbClient.schema.createTable('messages', table => {
+      table.uuid('id', { primaryKey: true }).defaultTo(dbClient.fn.uuid());
+      table.string('origin').notNullable();
+      table.timestamp('created').defaultTo(dbClient.fn.now());
+      table.string('title').notNullable();
+      table.text('message');
+      table.string('topic');
+      table.boolean('is_system').notNullable(); // is it a system message or a message for specific users and groups
+    });
+  }
+
+  if (!(await dbClient.schema.hasTable('users'))) {
+    await dbClient.schema.createTable('users', table => {
+      table.uuid('message_id').notNullable();
+      table.string('user').notNullable();
```

Comment:
@ydayagi should there be an index created for the user column since it is being used to query?

Reply:
https://issues.redhat.com/browse/FLPATH-727

Reply:
fixed